database.go

```
@@ -66,33 +66,26 @@ func getUser(userToSearch User) (User, error) {
 	return dbUser, result.Error
 }
 
-func postMessage(e Event, u User) {
-	e.InstanceId = u.InstanceId
-	e.UserId = u.UserId
-	e.EventTime = time.Now()
-	e.EventType = 1
-	dbcopy := db.Create(&e)
-	if dbcopy.Error != nil {
-		panic(dbcopy.Error)
+func updateUser(user User, updatedUser User) error {
+	result := db.Model(&user).Updates(&updatedUser)
+	if result.Error != nil {
+		panic(result.Error)
```

Why are we panicking here? With this control flow, the only possible return value is `nil`. Why not just return the error, regardless of whether it's `nil` or not, and let the caller decide how to handle it?